publish_courses/api.py

```diff
@@ -159,171 +146,97 @@ def create(self, request, *args, **kwargs) -> Response:
-        except Exception as e:
-            message = f'Error getting courses for account {account}'
-            if isinstance(e, CanvasAPIError):
-                message += ', SDK error details={}'.format(e)
-            logger.exception(message)
-            raise e
+        op_config = {
+            'account': account,
+            'term': term
+        }
+        op = BulkCourseSettingsOperation(op_config)
+        op.get_canvas_courses()
 
-        return canvas_courses
+        logger.info(f"Retrieved {len(op.canvas_courses)} courses for {account} and term {term}.")
+        return op.canvas_courses
     
-    def create_and_save_job(self, job_id: str, account_sis_id: str, audit_user_id: int,
-                            audit_user_name: str, selected_courses: List[int],
-                            canvas_courses: List[Dict]) -> None:
+    def create_and_save_job(self, account_sis_id: str, term: str, audit_user_id: int,
+                            audit_user_name: str, selected_courses: List[int]) -> Dict:
         """
-        Retrieves workflow states for selected courses, creates a job record,
-        and job details then adds them to the database.
+        Creates a job record, then creates job details then adds them to the database.
+        Returns a dictionary containing job and job details (course IDs and PKs).
         """
+        logger.info(f'Saving bulk publish courses job to database.')
+        
         # Create job record in the database.
-        job = Job.objects.create(job=job_id,
-                                 school_id=account_sis_id[len('school:'):],
+        job = Job.objects.create(school_id=account_sis_id[len('school:'):],
+                                 term_id=term,
                                  created_by_user_id=audit_user_id,
-                                 user_full_name=audit_user_name,
-                                 job_details_total_count=len(selected_courses))
+                                 user_full_name=audit_user_name)
         job.save()
+        logger.info(f'Bulk publish courses job saved to database. Job ID {job.pk}.')
 
-        logger.debug(f'Bulk publish courses job saved to database',
-                     extra={'job_id': job_id})
+        logger.info(f'Saving bulk publish courses job details for job ID {job.pk} to database.')
+        # Create JobDetails objects to efficiently insert all objects into the database in a single query.
+        job_details_objects = [JobDetails(parent_job=job, canvas_course_id=course_id) for course_id in selected_courses]
+        just_created_job_objects = JobDetails.objects.bulk_create(job_details_objects)  # Bulk create JobDetails objects (save to database).
+
+        # Construct the job dictionary (will be used by `send_job_to_queueing_lambda` func as payload).
+        job_dict = {
+            "job_id": job.pk,
+            "course_list": [{"course_id": jo.canvas_course_id, "job_detail_id": jo.pk} for jo in just_created_job_objects]
```

Comment:
Nit: Even though jo.pk should work since the `id` field is the PK for that model, I would lean towards having `jo.id`